microsoft/testsuites/display/modetest.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from typing import cast
+
+from lisa.executable import Tool
+from lisa.operating_system import Posix, Redhat, Ubuntu
+from lisa.tools.git import Git
+from lisa.util import UnsupportedDistroException
+
+
+class Modetest(Tool):
+    repo = "https://github.com/grate-driver/libdrm"
+
+    @property
+    def command(self) -> str:
+        return "modetest"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def is_status_connected(self, driver_name: str) -> bool:
+        cmd_result = self.run(f"-M {driver_name}", sudo=True, shell=True)
+        # output segment
+        # Connectors:
+        # id encoder status          name             size (mm)       modes   encoders
+        # 31 35      connected       Virtual-1        0x0             24      35
+        for line in cmd_result.stdout.splitlines():
+            if "connected" in line:
+                return True
+        return False
+
+    def _install(self) -> bool:
+        if isinstance(self.node.os, Ubuntu):
+            self.node.os.install_packages("libdrm-tests")
+        if isinstance(self.node.os, Redhat):
+            self._install_from_src()
+        return self._check_exists()
+
+    def _install_dep_packages(self) -> None:
+        posix_os: Posix = cast(Posix, self.node.os)
+        if isinstance(self.node.os, Redhat):
+            package_list = [
+                "git",
+                "make",
+                "autoconf",
+                "automake",
+                "libpciaccess-devel.x86_64",
+                "libtool",
+                "http://mirror.stream.centos.org/9-stream/CRB/x86_64/os/Packages/xorg-x11-util-macros-1.19.3-4.el9.noarch.rpm",  # noqa: E501
+                "http://mirror.stream.centos.org/9-stream/CRB/x86_64/os/Packages/ninja-build-1.10.2-6.el9.x86_64.rpm",  # noqa: E501
+                "http://mirror.stream.centos.org/9-stream/CRB/x86_64/os/Packages/meson-0.58.2-1.el9.noarch.rpm",  # noqa: E501
+            ]
+        else:
+            raise UnsupportedDistroException(self.node.os)
+        posix_os.install_packages(package_list)
```

Comment:
package_list shouldn't be a list if you're not planing on modifying it, but you also don't need to use a variable if all you're doing is passing it to one function.
Does mypy complain if you do it like this? I would think the conditional should give mypy context to know the object will have the `install_packages()` method.

```suggestion
        if isinstance(self.node.os, Redhat):
            self.node.os.install_packages(
                (
                    "git",
                     "make",
                     "autoconf",
                     "automake",
                     "libpciaccess-devel.x86_64",
                     "libtool",
                     "http://mirror.stream.centos.org/9-stream/CRB/x86_64/os/Packages/xorg-x11-util-macros-1.19.3-4.el9.noarch.rpm",  # noqa: E501
                     "http://mirror.stream.centos.org/9-stream/CRB/x86_64/os/Packages/ninja-build-1.10.2-6.el9.x86_64.rpm",  # noqa: E501
                     "http://mirror.stream.centos.org/9-stream/CRB/x86_64/os/Packages/meson-0.58.2-1.el9.noarch.rpm",  # noqa: E501
                )
            )
        else:
            raise UnsupportedDistroException(self.node.os)
```